core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -1852,7 +1922,7 @@ public final void truncate() {
         }
 
         if (size() == 0) {
-            return;
+            return -1;
```

Comment:
NO_TXN is used as a return value to tell the caller that a new txn number cannot be provided because the table structure is stale and needs to be updated.
NO_TXN value is Long.MIN_VALUE, not -1.
However, in this case return value is not used so truncate() can be void actually.
Changed the signature.